lib/improver/nbhood.py

```diff
@@ -55,107 +59,243 @@ class BasicNeighbourhoodProcessing(object):
+            )
         if (grid_cells_x > self.MAX_KERNEL_CELL_RADIUS or
                 grid_cells_y > self.MAX_KERNEL_CELL_RADIUS):
             raise ValueError(
                 ("Neighbourhood processing radius of " +
-                 "{0} km ".format(self.radius_in_km) +
+                 "{0} km ".format(radii_in_km) +
                  "exceeds maximum grid cell extent")
             )
         return grid_cells_x, grid_cells_y
 
-    def process(self, cube):
+    def _apply_kernel_for_smoothing(self, cube, ranges):
         """
-        Set the specified name and units metadata to the cube from the upstream
-        plugin.
+        Return the number of grid cells in the x and y direction
+        to be used to create the kernel.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube containing the x and y coordinates, which will be used for
+            calculating the number of grid cells in the x and y direction,
+            which equates to the size of the desired radii.
+        ranges : Tuple
+            Number of grid cells in the x and y direction used to create
+            the kernel.
 
         Returns
         -------
-        Cube
-            The cube from the upstream plugin with name and units metadata
+        cube : Iris.cube.Cube
+            Cube containing the smoothed field after the kernel has been
             applied.
 
         """
-        try:
-            realiz_coord = cube.coord('realization')
-        except iris.exceptions.CoordinateNotFoundError:
-            pass
-        else:
-            if len(realiz_coord.points) > 1:
-                raise ValueError("Does not operate across realizations.")
-        if np.isnan(cube.data).any():
-            raise ValueError("Error: NaN detected in input cube data")
         data = cube.data
-        ranges = self.get_grid_x_y_kernel_ranges(cube)
-        fullranges = np.zeros([np.rank(data)])
+        fullranges = np.zeros([np.ndim(data)])
         axes = []
-        for coord_name in ['projection_x_coordinate',
-                           'projection_y_coordinate']:
-            axes.append(cube.coord_dims(coord_name)[0])
+        try:
+            for coord_name in ['projection_x_coordinate',
+                               'projection_y_coordinate']:
+                axes.append(cube.coord_dims(coord_name)[0])
+        except CoordinateNotFoundError:
+            raise ValueError("Invalid grid: projection_x/y coords required")
         for axis_index, axis in enumerate(axes):
             fullranges[axis] = ranges[axis_index]
-        kernel = np.ones([1 + x * 2 for x in fullranges])
-        n = np.ogrid[tuple([slice(-x, x+1) for x in ranges])]
+        # Define the size of the kernel based on the number of grid cells
+        # contained within the desired radius.
+        kernel = np.ones([int(1 + x * 2) for x in fullranges])
+        # Create an open multi-dimensional meshgrid.
```

Comment:
Is it wise to use the name meshgrid when np.meshgrid is commonly employed? It's a judgement and I shall leave it to you.